dds/DCPS/TopicImpl.cpp

```diff
@@ -68,19 +67,17 @@ TopicImpl::set_qos(const DDS::TopicQos & qos)
 
     } else {
       qos_ = qos;
-      DomainParticipantImpl* part =
-        dynamic_cast<DomainParticipantImpl*>(this->participant_);
 
       Discovery_rch disco =
-        TheServiceParticipant->get_discovery(part->get_domain_id());
+        TheServiceParticipant->get_discovery(this->participant_->get_domain_id());
```

Comment:
Please keep in mind the coding guideline "this-> is not used unless required for disambiguation or to access members of a template-dependent base class."  I don't think you need to go back and change these files just for this issue, but if you end up changing them anyway it would be good to bring them up to date with the guideline.